lib/dotcom/trip_plan/itinerary_groups.ex

```diff
@@ -0,0 +1,145 @@
+defmodule Dotcom.TripPlan.ItineraryGroups do
+  @moduledoc """
+  Group itineraries by unique legs.
+
+  A unique leg is defined as a leg that has a unique combination of mode, from, and to.
+  But, this does not include walking legs that are less than 0.2 miles.
+  """
+
+  alias Dotcom.TripPlan.{Itinerary, Leg, PersonalDetail, TransitDetail}
+  alias OpenTripPlannerClient.ItineraryTag
+
+  @spec from_itineraries([Itinerary.t()]) :: [
+          %{itineraries: [Itinerary.t()], summary: map()}
+        ]
+  def from_itineraries(itineraries) do
+    itineraries
+    |> Enum.group_by(&unique_legs_to_hash/1)
+    |> Enum.map(&group_departures/1)
+    |> Enum.reject(&Enum.empty?(&1))
+    |> Enum.sort_by(fn
+      %{itineraries: [%{tag: tag} | _] = _} ->
+        Enum.find_index(ItineraryTag.tag_priority_order(), &(&1 == tag))
+
+      _ ->
+        -1
+    end)
+  end
+
+  defp unique_legs_to_hash(%Itinerary{legs: legs}) do
+    legs
+    |> Enum.reject(&short_walking_leg?/1)
+    |> Enum.map(&unique_leg_to_tuple/1)
+    |> :erlang.phash2()
+  end
+
+  defp unique_leg_to_tuple(%Leg{mode: %PersonalDetail{}} = leg) do
+    {:WALK, leg.from.name, leg.to.name}
+  end
+
+  defp unique_leg_to_tuple(%Leg{mode: %{route: route}} = leg) do
+    {Routes.Route.type_atom(route.type), leg.from.name, leg.to.name}
+  end
+
+  defp group_departures({_hash, grouped_itineraries}) do
+    summarized_legs =
+      grouped_itineraries
+      |> Enum.map(& &1.legs)
+      |> Enum.zip_with(&Function.identity/1)
```

Comment:
```suggestion
      |> Enum.zip()
```

I might be misunderstanding the [Enum.zip/1 docs](https://hexdocs.pm/elixir/1.12/Enum.html#zip/1), but I think this would do the same thing?

Reply:
Huh, interesting - almost the same!

```elixir
iex(1)> [[1, 2, 3],[:a, :b, :c]] |> Enum.zip()
[{1, :a}, {2, :b}, {3, :c}]
iex(2)> [[1, 2, 3],[:a, :b, :c]] |> Enum.zip_with(&Function.identity/1)
[[1, :a], [2, :b], [3, :c]]
```

I think I'll keep it the way it is since it saves me the step of converting the tuples to enmuerables.